### PR TITLE
Improve TypeScript config

### PR DESCRIPTION
**Pull request by valtlai**
- Upgrades to the latest, major TypeScript version
- Removes the unused, extra types file

TSConfig:

- Updates `target` and `module` to modern but Node.js 18 compatible values
- Removes the following options:
  - `esModuleInterop` and `allowSyntheticDefaultImports` (which have default values)
  - `moduleResolution` (which conflicts with `"module": "nodenext"`, and whose default value is taken from `module`)
  - `checkJs` (because there are no .js files, only .ts ones)
  - `exclude` (because the listed directories are not inside `include` and thus not included by default)
- Simplifies the `include` list to use just the source directory

### CODE REVIEW

File: src/db/index_knex.ts [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import dotenv from 'dotenv';
-import knex from 'knex';
+
+// Work around the incompatibility between Knex typings
+// and the "module": "nodenext" TS config
+// Will be fixed once a new version of Knex with the following PR is released:
+// https://github.com/knex/knex/pull/5659
+import * as knexModule from 'knex';
+const { knex } = knexModule.default;
```
Comment (valtlai) [inline src/db/index_knex.ts:8]: A new Knex version fixing this should be released tonight – https://github.com/knex/knex/pull/5659#issuecomment-1742723386 – so no need to merge the PR yet.

**Review by valju — COMMENTED** [review]
Good work! And this is fixing something I had troubles two weeks ago, when trying to make Knex work fully TypeScript. Nice to hear fix arrives.

**Comment by valju** [comment]
How did the Knex people do, they have an update for this? Is this already final? When should we accept this?

**Comment by valtlai** [comment]
They have an update, yeah, but it’s not released yet. I don’t know when it will happen. Let’s merge this now, and I’ll open a new PR later once the new Knex version is available.

**Review by valju — APPROVED** [review]

**Pull request merged**